.github/workflows/main.yml

```diff
@@ -186,3 +196,78 @@ jobs:
         name: taisei_windows_fail_log
         path: build/meson-logs/meson-log.txt
         if-no-files-found: warn
+
+  emscripten-test-build:
+    name: Emscripten Test Build (WebGL)
+    if: "!contains(github.event.head_commit.message, '[skip ci]')"
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
+
+    - name: Set Up Build Environment
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Install Build Tools
+      run: >
+        sudo apt update
+
+        sudo apt install -y -qq
+        python3-docutils
+        python3-pip
+        git
+
+        pip install
+        meson==0.56.2
+        ninja
+        zstandard
+
+    - name: Pull cached emsdk
+      id: cache-emsdk
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.EM_CACHE_FOLDER }}
+        key: ${{ env.EM_VERSION }}-${{ runner.os }}
```

Comment:
The cache mechanism is interesting. Maybe we could build and install the native shader tools the same way (spirv-cross and glslc)? Those probably take more time to build than Taisei itself. Doesn't have to be done in this PR, though.

Reply:
It's a bit trickier to pull off with things inside the `build/` directory simply because of how picky `meson` is about build caches. It may or may not save time in the end.

But yeah, let's do that in another PR.

Reply:
It's not necessary (or recommended) to mess with the build directory here. We can just install the correct versions of the tools into an isolated (cached) directory, disable the `force_vendored_shader_tools` option, and point Meson to the prebuilt binaries (either via `PATH` manipulation or explicitly in machine files)